pkg/operator/ceph/cluster/utils.go

```diff
@@ -30,28 +30,84 @@ import (
 
 // populateConfigOverrideConfigMap creates the "rook-config-override" config map
 // Its content allows modifying Ceph configuration flags
-func populateConfigOverrideConfigMap(clusterdContext *clusterd.Context, namespace string, ownerInfo *k8sutil.OwnerInfo) error {
+func populateConfigOverrideConfigMap(clusterdContext *clusterd.Context, namespace string, ownerInfo *k8sutil.OwnerInfo, clusterMetadata metav1.ObjectMeta) error {
 	ctx := context.TODO()
-	placeholderConfig := map[string]string{
-		k8sutil.ConfigOverrideVal: "",
-	}
 
-	cm := &v1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      k8sutil.ConfigOverrideName,
-			Namespace: namespace,
-		},
-		Data: placeholderConfig,
+	existingCM, err := clusterdContext.Clientset.CoreV1().ConfigMaps(namespace).Get(ctx, k8sutil.ConfigOverrideName, metav1.GetOptions{})
+	if err != nil {
+		if !kerrors.IsNotFound(err) {
+			logger.Warningf("failed to get cm %q to check labels and annotations", k8sutil.ConfigOverrideName)
```

Comment:
```suggestion
			logger.Warningf("failed to get cm %q to check labels and annotations", k8sutil.ConfigOverrideName)
```
should we also print `err` with logs?

Reply:
This is really handling a corner scenario so I preferred to just log the issue and not fail the reconcile. Adding the labels only ever needs to happen once, so it's highly unlikely to fail during that one time, but we still try to read the configmap with each reconcile.